hail/src/main/scala/is/hail/expr/ir/lowering/LoweringPass.scala

```diff
@@ -54,4 +56,40 @@ case object LowerTableToDistributedArrayPass extends LoweringPass {
   val context: String = "LowerTableToDistributedArray"
 
   def transform(ctx: ExecuteContext, ir: BaseIR): BaseIR = LowerTableIR.lower(ir.asInstanceOf[IR])
+}
+
+case object LowerArrayAggsToRunAggs extends LoweringPass {
+  val before: IRState = CompilableIR
+  val after: IRState = EmittableIR
+  val context: String = "LowerArrayAggsToRunAggs"
+
+  def transform(ctx: ExecuteContext, ir: BaseIR): BaseIR = RewriteBottomUp(ir, {
+    case ArrayAgg(a, name, query) =>
+      val res = genUID()
+      val aggs = Extract(query, res)
+      Some(Let(
```

Comment:
Neat that we can rewrite the previous code (found in Emit) this tersely by using our existing primitives